### PR TITLE
Enhancement/refactor js files

### LINKED ISSUE #23: Refactor js files
Now Typescript is configurated, refactor code using Typescript features and change file extension from .js to .ts

### PR DESCRIPTION
**Pull request by KsquarianMike**
Closes #23

### CODE REVIEW

**Comment by KsquarianMike** [comment]
i'am using a lot of variables type "any", when the "mappers" be implemented I will change it

File: api/app.ts [hunk 1]
```diff
@@ -1,51 +1,29 @@
...
-/**
- * project libraries
- */
+} from "./middlewares/errorHandlers";
+import routes from "./routes/v1";
 
-const utils = require("./utils");
-const response = require("./utils/response");
+const debug = Debug("chef:orders:app");
+const slash = require("express-slash");
```
Comment (mishelashala) [inline api/app.ts]: shouldn't this be imported, not required?

File: api/app.ts [hunk 2]
```diff
@@ -69,20 +47,21 @@ app.use(bodyParser.json());
 
 // routes
 routes(app);
+
 // v1/name/ redirect to v1/name
 app.use(slash());
-app.use(function(req, res, next) {
+
+app.use(function(req: Request, res: Response) {
```
Comment (mishelashala) [inline api/app.ts]: favor `arrow` functions over `anonymous functions`

File: api/app.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import * as Debug from "debug";
 import * as helmet from "helmet";
 import * as express from "express";
 import * as bodyParser from "body-parser";
+import slash = require("express-slash");
```
Comment (mishelashala) [inline api/app.ts]: wut, doesn't it has support for `import * as slash from 'express-slash'`?

**Pull request merged**